libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1652,27 +1681,11 @@ IotMqttError_t IotMqtt_PublishAsync( IotMqttConnection_t mqttConnection,
     _mqttOperation_t * pOperation = NULL;
     uint8_t ** pPacketIdentifierHigh = NULL;
 
-    /* Check that IotMqtt_Init was called. */
-    if( _checkInit() == false )
-    {
-        status = IOT_MQTT_NOT_INITIALIZED;
-    }
-    else if( _IotMqtt_ValidatePublish( mqttConnection->awsIotMqttMode,
-                                       pPublishInfo,
+    /* Create a PUBLISH operation. */
```

Comment:
Better to have asserts in the internal function to validate the arguments.

Reply:
Added assert checks.